runway/blueprints/staticsite/auth_at_edge.py

```diff
@@ -0,0 +1,428 @@
+                FunctionName=lambda_function.ref()
+            )
+        )
+
+    def add_lambda_execution_role(self):
+        # type: () -> iam.Role
+        """Create the Lambda@Edge execution role."""
+        return self.template.add_resource(
+            iam.Role(
+                'LambdaExecutionRole',
+                AssumeRolePolicyDocument=make_simple_assume_policy(
+                    'lambda.amazonaws.com', 'edgelambda.amazonaws.com'
+                ),
+                ManagedPolicyArns=[
+                    self.IAM_ARN_PREFIX + 'AWSLambdaBasicExecutionRole'
+                ]
+            )
+        )
+
+    def get_distribution_options(self,
+                                 bucket,  # type: s3.Bucket
+                                 oai,  # type: cloudfront.CloudFrontOriginAccessIdentity
+                                 lambda_funcs,  # type: List[cloudfront.LambdaFunctionAssociation]
+                                 check_auth_lambda_version,  # type: awslambda.Version
+                                 http_headers_lambda_version,  # type: awslambda.Version
+                                 parse_auth_lambda_version,  # type: awslambda.Version
+                                 refresh_auth_lambda_version,  # type: awslambda.Version
+                                 sign_out_lambda_version  # type: awslambda.Version
+                                ):  # noqa: E124
+        # type: (...) -> Dict[str, Any]
+        """Retrieve the options for our CloudFront distribution.
+
+        Keyword Args:
+            bucket (dict): The bucket resource
+            oai (dict): The origin access identity resource
+
+        Return:
+            dict: The CloudFront Distribution Options
+
+        """
+        variables = self.get_variables()
+
+        default_cache_behavior_lambdas = lambda_funcs
+        default_cache_behavior_lambdas.append(
+            cloudfront.LambdaFunctionAssociation(
+                EventType='viewer-request',
+                LambdaFunctionARN=check_auth_lambda_version.ref()
+            )
+        )
+        default_cache_behavior_lambdas.append(
+            cloudfront.LambdaFunctionAssociation(
+                EventType='origin-response',
+                LambdaFunctionARN=http_headers_lambda_version.ref()
+            )
+        )
+
+        return {
+            'Aliases': self.add_aliases(),
+            'Origins': [
+                cloudfront.Origin(
+                    DomainName=Join(
+                        '.',
+                        [bucket.ref(),
+                         's3.amazonaws.com']),
+                    S3OriginConfig=cloudfront.S3OriginConfig(
+                        OriginAccessIdentity=Join(
+                            '',
+                            ['origin-access-identity/cloudfront/',
+                             oai.ref()])
+                    ),
+                    Id='protected-bucket'
+                ),
+                cloudfront.Origin(
```

Comment:
Does the login flow actually require this secondary dummy-origin? Looking at the behaviors, I'm wondering if they can just be pointed at the main s3 origin (since they're all just jumping in front of the flow via viewer-request associations)?

Reply:
https://github.com/aws-samples/cloudfront-authorization-at-edge/issues/35

Removed for now, have a question out to the originator

Reply:
Still waiting on request from originator, but all my testing seems to fall inline with what you're saying here. Going to mark this as resolved.